engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/PassPersistance.java

```diff
@@ -0,0 +1,155 @@
+    protected TailCall.TailPosition readObject(Input in)
+        throws IOException, ClassNotFoundException {
+      var b = in.readBoolean();
+      return b
+          ? org.enso.compiler.pass.analyse.TailCall$TailPosition$Tail$.MODULE$
+          : org.enso.compiler.pass.analyse.TailCall$TailPosition$NotTail$.MODULE$;
+    }
+  }
+
+  @org.openide.util.lookup.ServiceProvider(service = Persistance.class)
+  public static final class PersistAliasAnalysisGraphScope
+      extends Persistance<org.enso.compiler.pass.analyse.AliasAnalysis$Graph$Scope> {
+    public PersistAliasAnalysisGraphScope() {
+      super(org.enso.compiler.pass.analyse.AliasAnalysis$Graph$Scope.class, false, 1124);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected org.enso.compiler.pass.analyse.AliasAnalysis$Graph$Scope readObject(Input in)
+        throws IOException {
+      var childScopes = in.readInline(scala.collection.immutable.List.class);
+      var occurrences = (scala.collection.immutable.Set) in.readObject();
+      var allDefinitions = in.readInline(scala.collection.immutable.List.class);
+      var parent =
+          new org.enso.compiler.pass.analyse.AliasAnalysis$Graph$Scope(
+              childScopes, occurrences, allDefinitions);
+      var optionParent = Option.apply(parent);
+      childScopes.forall(
+          (object) -> {
+            var ch = (org.enso.compiler.pass.analyse.AliasAnalysis$Graph$Scope) object;
+            ch.parent_$eq(optionParent);
+            return null;
+          });
+      return parent;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected void writeObject(
+        org.enso.compiler.pass.analyse.AliasAnalysis$Graph$Scope obj, Output out)
+        throws IOException {
+      out.writeInline(scala.collection.immutable.List.class, obj.childScopes());
+      out.writeObject(obj.occurrences());
+      out.writeInline(scala.collection.immutable.List.class, obj.allDefinitions());
+    }
+  }
+
+  @org.openide.util.lookup.ServiceProvider(service = Persistance.class)
+  public static final class PersistAliasAnalysisGraph extends Persistance<Graph> {
+    public PersistAliasAnalysisGraph() {
+      super(Graph.class, false, 1119);
+    }
+
+    @SuppressWarnings("unchecked")
+    protected Graph readObject(Input in) throws IOException {
+      var g = new Graph();
+
+      var rootScope = (AliasAnalysis$Graph$Scope) in.readObject();
+      assignParents(rootScope);
+      g.rootScope_$eq(rootScope);
+
+      var links =
+          (scala.collection.immutable.Set) in.readInline(scala.collection.immutable.Set.class);
+      g.links_$eq(links);
+      return g;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    protected void writeObject(Graph obj, Output out) throws IOException {
+      out.writeObject(obj.rootScope());
+      out.writeInline(scala.collection.immutable.Set.class, obj.links());
+    }
+
+    private static void assignParents(AliasAnalysis$Graph$Scope scope) {
```

Comment:
Could probably be shared with `AliasAnalysis$Graph$Scope`?

Reply:
I don't understand this comment.